packages/commonwealth/client/scripts/views/components/component_kit/CWCoverImageUploader/UploadControl.tsx

```diff
@@ -0,0 +1,255 @@
+import clsx from 'clsx';
+import React from 'react';
+import { CWIcon } from '../cw_icons/cw_icon';
+import { CWText } from '../cw_text';
+import { CWButton } from '../new_designs/CWButton';
+import CWCircleMultiplySpinner from '../new_designs/CWCircleMultiplySpinner';
+import CWIconButton from '../new_designs/CWIconButton';
+import { CWTextInput } from '../new_designs/CWTextInput';
+import { MessageRow } from '../new_designs/CWTextInput/MessageRow';
+import { ImageBehavior, UploadControlProps } from './types';
+import { useUploadControl } from './useUploadControl';
+
+export const UploadControl = ({
+  name,
+  hookToForm,
+  imageURL,
+  withAIImageGeneration,
+  disabled,
+  imageBehavior = ImageBehavior.Circle,
+  uploadControlClassName,
+  canSwitchBetweenProcessedImages,
+  processedImages: providedProcessedImages,
+  onImageProcessingChange,
+  onImageGenerated,
+  onImageUploaded,
+  onProcessedImagesListChange,
+}: UploadControlProps) => {
+  const {
+    areActionsDisabled,
+    isLoading,
+    isDraggingFile,
+    formFieldErrorMessage,
+    imageToRender,
+    openFilePicker,
+    registeredFormContext,
+    dropzoneRef,
+    imageInputRef,
+    setImageInputRefUpdated,
+    processedImages,
+    activeImageIndex,
+    isWindowExtraSmall,
+    updateActiveImageIndex,
+    isHookedToForm,
+    placeholder,
+    setIsImageGenerationSectionOpen,
+    setImagePrompt,
+    isImageGenerationSectionOpen,
+    imagePrompt,
+    generateImage,
+  } = useUploadControl({
+    name,
+    hookToForm,
+    imageURL,
+    withAIImageGeneration,
+    disabled,
+    imageBehavior,
+    uploadControlClassName,
+    canSwitchBetweenProcessedImages,
+    processedImages: providedProcessedImages,
+    onImageProcessingChange,
+    onImageGenerated,
+    onImageUploaded,
+    onProcessedImagesListChange,
+  });
+
+  return (
+    <div
+      role="button"
+      tabIndex={0}
+      className={clsx(
+        'UploadControl',
+        { disabled: areActionsDisabled },
+        { isLoading },
+        { hovered: isDraggingFile },
+        { formError: !!formFieldErrorMessage },
+        { hasImageURL: !!imageToRender },
+        uploadControlClassName,
+      )}
+      onClick={() => openFilePicker()}
```

Comment:
nit: if no additional param passed it could just be 

`      onClick={openFilePicker}`